tests/integrationv2/providers.py

```diff
@@ -159,15 +159,15 @@ def setup_client(self):
         if self.options.protocol == Protocols.TLS13:
             cmd_line.append('--tls13')
 
-        if self.options.cipher is not None:
-            if self.options.cipher is Ciphers.KMS_PQ_TLS_1_0_2019_06:
-                cmd_line.extend(['-c', 'KMS-PQ-TLS-1-0-2019-06'])
-            elif self.options.cipher is Ciphers.PQ_SIKE_TEST_TLS_1_0_2019_11:
-                cmd_line.extend(['-c', 'PQ-SIKE-TEST-TLS-1-0-2019-11'])
-            else:
-                cmd_line.extend(['-c', 'test_all'])
-        else:
-            cmd_line.extend(['-c', 'test_all'])
+        cipher_prefs = 'test_all_tls12'
+        if self.options.cipher is Ciphers.KMS_PQ_TLS_1_0_2019_06:
+            cipher_prefs = 'KMS-PQ-TLS-1-0-2019-06'
+        elif self.options.cipher is Ciphers.PQ_SIKE_TEST_TLS_1_0_2019_11:
+            cipher_prefs = 'PQ-SIKE-TEST-TLS-1-0-2019-11'
+        elif self.options.protocol is Protocols.TLS13:
+            cipher_prefs = 'test_all'
```

Comment:
Does this change work for the happy path test? This may simplify the `valid_test` fixture. I remember tls13 used to fail with S2N client + server with `-c test_all` because the RC4 cipher would be used (and, of course, fail). Now that tls12 and tls13 are separate that may be fixed.

This change looks fine though, just interested.

Reply:
"test_all_tls12" still contains RC4, and is still ordered by IANA value rather than by preference, so you would still have a problem with S2N TLS1.2 selecting RC4. The only difference between "test_all_tls12" and "test_all" is the missing TLS1.3 cipher suites.